### PR TITLE
feat(metaserver): support get pod by uid

### PR DESCRIPTION
**Pull request by cheney-lin**
#### What type of PR is this?

Enhancements

### CODE REVIEW

**Review by luomingmeng — DISMISSED** [review]

**Comment by waynepeking348** [comment]
pod-fetcher interface is kind of mess and should be refined before it can be approved cc @luomingmeng

File: pkg/metaserver/agent/pod/kubelet.go [hunk 1]
```diff
@@ -30,23 +30,34 @@ type KubeletPodFetcher interface {
 	// GetPodList returns those latest pods, and podFilter is a function to filter a pod,
 	// if pod passed return true else return false, if podFilter is nil return all pods
 	GetPodList(ctx context.Context, podFilter func(*v1.Pod) bool) ([]*v1.Pod, error)
+
+	// GetPod returns Pod by UID
+	GetPod(ctx context.Context, podUID string) (*v1.Pod, error)
```
Comment (luomingmeng) [inline pkg/metaserver/agent/pod/kubelet.go:35]: KubeletPodFetcher is underlay interface to get k8s pod, only need provide get raw pod list function, and no need provide `GetPod` function to get pod by uid，the `GetPod` function should provide in `PodFetcher`

File: pkg/metaserver/agent/pod/runtime.go [hunk 2]
```diff
@@ -55,23 +55,27 @@ type labeledContainerInfo struct {
...
 
-	return &RuntimePodFetcher{runtimeService: runtimeService}, nil
+	return &RuntimePodFetcherImpl{runtimeService: runtimeService}, nil
 }
 
 // GetPods returns a list of containers grouped by pods. The boolean parameter
 // specifies whether the runtime returns all containers including those already
 // exited and dead containers (used for garbage collection).
-func (r *RuntimePodFetcher) GetPods(all bool) ([]*RuntimePod, error) {
+func (r *RuntimePodFetcherImpl) GetPods(all bool) ([]*RuntimePod, error) {
```
Comment (luomingmeng) [inline pkg/metaserver/agent/pod/runtime.go:78]: this struct can be private, the same as `kubeletPodFetcherImpl`

File: pkg/metaserver/agent/pod/runtime.go [hunk 3]
```diff
@@ -55,23 +55,27 @@ type labeledContainerInfo struct {
...
+type RuntimePodFetcher interface {
+	GetPods(all bool) ([]*RuntimePod, error)
+}
+
+type RuntimePodFetcherImpl struct {
 	runtimeService cri.RuntimeService
 }
 
-func NewRuntimePodFetcher(conf *config.Configuration) (*RuntimePodFetcher, error) {
+func NewRuntimePodFetcher(conf *config.Configuration) (*RuntimePodFetcherImpl, error) {
```
Comment (luomingmeng) [inline pkg/metaserver/agent/pod/runtime.go:66]: construct function return interface `RuntimePodFetcher` directly to avoid mistake use

Comment (cheney-lin) [reply pkg/metaserver/agent/pod/runtime.go]: fixed

**Review by luomingmeng — DISMISSED** [review]

File: pkg/metaserver/agent/pod/kubelet.go [hunk 4]
```diff
@@ -35,18 +35,26 @@ type KubeletPodFetcher interface {
 // kubeletPodFetcherImpl use kubelet 10255 pods interface to get pod directly without cache.
 type kubeletPodFetcherImpl struct{}
 
-// GetPodList get pods from kubelet 10255/pods api, and the returned slice does not
-// contain pods that don't pass `podFilter`
-func (k *kubeletPodFetcherImpl) GetPodList(_ context.Context, podFilter func(*v1.Pod) bool) ([]*v1.Pod, error) {
+func getPodsByKubeletAPI() (v1.PodList, error) {
```
Comment (waynepeking348) [inline pkg/metaserver/agent/pod/kubelet.go:38]: why we need to extract this function out?

Comment (cheney-lin) [reply pkg/metaserver/agent/pod/kubelet.go]: No need any more, removed already.

**Review by waynepeking348 — APPROVED** [review]

**Review by luomingmeng — APPROVED** [review]

**Pull request merged**